product_variant_configurator/models/product_configurator.py

```diff
@@ -0,0 +1,268 @@
+        inverse_name='owner_id', string='Product attributes', copy=True)
+    price_extra = fields.Float(
+        compute='_compute_price_extra',
+        digits=dp.get_precision('Product Price'),
+        help="Price Extra: Extra price for the variant with the currently "
+             "selected attributes values on sale price. eg. 200 price extra, "
+             "1000 + 200 = 1200.")
+    product_id = fields.Many2one(
+        string="Product",
+        comodel_name="product.product")
+    name = fields.Char()
+    can_create_product = fields.Boolean(
+        compute='_compute_can_be_created',
+        store=False)
+    create_product_variant = fields.Boolean(
+        string="Create product now!")
+
+    @api.multi
+    @api.depends('product_attribute_ids', 'product_attribute_ids.value_id',
+                 'product_id')
+    def _compute_can_be_created(self):
+        for rec in self:
+            if rec.product_id:
+                # product already selected
+                rec.can_create_product = False
+                continue
+            if not rec.product_tmpl_id:
+                # no product nor template
+                rec.can_create_product = False
+                continue
+            rec.can_create_product = not bool(
+                len(rec.product_tmpl_id.attribute_line_ids.mapped(
+                    'attribute_id')) -
+                len(filter(None,
+                           rec.product_attribute_ids.mapped('value_id'))))
+
+    @api.multi
+    @api.depends('product_attribute_ids', 'product_attribute_ids.value_id')
+    def _compute_price_extra(self):
+        for record in self:
+            record.price_extra = sum(
+                record.mapped('product_attribute_ids.price_extra'))
+
+    def _set_product_tmpl_attributes(self):
+        if self.product_tmpl_id:
+            attribute_lines = self.product_attribute_ids.browse([])
+            for attribute_line in self.product_tmpl_id.attribute_line_ids:
+                attribute_lines += attribute_lines.new({
+                    'attribute_id': attribute_line.attribute_id.id,
+                    'product_tmpl_id': self.product_tmpl_id.id,
+                    'owner_model': self._name,
+                    'owner_id': self.id,
+                })
+            self.product_attribute_ids = attribute_lines
+
+    def _set_product_attributes(self):
+        if self.product_id:
+            attribute_lines = self.product_attribute_ids.browse([])
+            for vals in self.product_id._get_product_attributes_values_dict():
+                vals['product_tmpl_id'] = self.product_id.product_tmpl_id
+                vals['owner_model'] = self._name
+                vals['owner_id'] = self.id
+                attribute_lines += attribute_lines.new(vals)
+            self.product_attribute_ids = attribute_lines
+
+    def _empty_attributes(self):
+        self.product_attribute_ids = self.product_attribute_ids.browse([])
```

Comment:
As an aside, isn't it cleaner just to use self.env['product.configurator.attribute'] - that gives an empty recordset and more conventional.